File: example/app/src/main/java/com/theah64/retrokitexample/adapters/MoviesAdapter.java
package com.theah64.retrokitexample.adapters;

import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.theah64.retrokit.adapters.BaseButterknifeRecyclerViewHolder;
import com.theah64.retrokit.adapters.BaseRecyclerViewAdapter;
import com.theah64.retrokitexample.R;
import com.theah64.retrokitexample.model.Movie;

import java.util.List;

import butterknife.BindView;

/**
 * Created by theapache64 on 26/8/17.
 */

public class MoviesAdapter extends BaseRecyclerViewAdapter<MoviesAdapter.ViewHolder, Movie> {

    public MoviesAdapter(List<Movie> data, @Nullable Callback<Movie> callback) {
        super(data, callback);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position, Movie item) {
        holder.tvMovieTitle.setText(item.getTitle());
        holder.tvMovieDescription.setText(item.getDescription());
    }

    @Override
    protected int getRowLayoutID() {
        return R.layout.movie_row;
    }

    @Override
    protected ViewHolder getNewRow(View row) {
        return new ViewHolder(row, this);
    }

    class ViewHolder extends BaseButterknifeRecyclerViewHolder<Movie> {

        @BindView(R.id.tvMovieTitle)
        TextView tvMovieTitle;

        @BindView(R.id.tvMovieDescription)
        TextView tvMovieDescription;

        public ViewHolder(View row, BaseRecyclerViewAdapter adapter) {
            super(row, adapter);
        }
    }
}

File: app/src/main/java/com/theah64/retrokitexample/adapters/MoviesAdapter.kt
package com.theah64.retrokitexample.adapters

import android.view.View
import android.widget.TextView
import butterknife.BindView
import com.theah64.retrokit.adapters.BaseButterKnifeRecyclerViewHolder
import com.theah64.retrokit.adapters.BaseRecyclerViewAdapter
import com.theah64.retrokitexample.R
import com.theah64.retrokitexample.model.Movie

/**
 * Created by theapache64 on 26/8/17.
 */

class MoviesAdapter(data: List<Movie>, callback: BaseRecyclerViewAdapter.Callback<Movie>?) : BaseRecyclerViewAdapter<MoviesAdapter.ViewHolder, Movie>(data, callback) {

    override fun onBindViewHolder(holder: ViewHolder, position: Int, item: Movie) {
        holder.tvMovieTitle!!.text = item.title
        holder.tvMovieDescription!!.text = item.description
    }

    override fun getRowLayoutID(): Int {
        return R.layout.movie_row
    }

    override fun getNewRow(row: View): ViewHolder {
        return ViewHolder(row, this)
    }

    inner class ViewHolder(row: View, adapter: BaseRecyclerViewAdapter<*, *>) : BaseButterKnifeRecyclerViewHolder<Movie>(row, adapter) {

        @BindView(R.id.tvMovieTitle)
        lateinit var tvMovieTitle: TextView

        @BindView(R.id.tvMovieDescription)
        lateinit var tvMovieDescription: TextView
    }
}
